server/routes/linkTokens.js

```diff
@@ -25,75 +26,86 @@ const redirect_uri =
+        const transIntentCreateRequest = {
+          client_id: PLAID_CLIENT_ID,
+          secret: PLAID_SECRET_SANDBOX,
+          mode: 'PAYMENT',
+          amount: subscriptionAmount.toFixed(2),
+          ach_class: 'ppd',
+          description: 'payment', // cannot be longer than 8 characters
+          user: {
+            legal_name: username,
+          },
+        };
+
+        const transferIntentCreateResponse = await plaid.transferIntentCreate(
+          transIntentCreateRequest
+        );
+        const transferIntentId =
+          transferIntentCreateResponse.data.transfer_intent.id;
+        // create new Transfer now so that you can reference its transferIntentId upon link success
+        // because the link success metadata does not pass back any data about the transfer except for transfer_status
+        const newTransfer = await createTransfer(
+          null, // item_id
+          userId,
+          null, // plaid_account_id
+          transferIntentId,
+          null, // authorization_id - for TransferUI transfers
+          null, // transfer_id
+          subscriptionAmount.toFixed(2),
+          null, // status
+          'debit',
+          null // sweep_status
+        );
+        return transferIntentId;
       };
-      let transferIntentId;
 
-      const transferIntentCreateResponse = await plaid.transferIntentCreate(
-        transIntentCreateRequest
-      );
-      transferIntentId = transferIntentCreateResponse.data.transfer_intent.id;
-      // create new Transfer now so that you can reference its transferIntentId upon link success
-      // because the link success metadata does not pass back any data about the transfer except for transfer_status
-      const newTransfer = await createTransfer(
-        null, // item_id
-        userId,
-        null, // plaid_account_id
-        transferIntentId,
-        null, // authorization_id - for TransferUI transfers
-        null, // transfer_id
-        subscriptionAmount.toFixed(2),
-        null, // status
-        'debit',
-        null // sweep_status
-      );
+      // create link token using transfer intent id
+      const createLinkTokenForTransfer = async () => {
+        let products = ['transfer'];
+        const response = await fetch('http://ngrok:4040/api/tunnels');
+        const { tunnels } = await response.json();
+        const httpTunnel = tunnels.find(t => t.proto === 'http');
+        const linkTokenParams = {
+          client_id: PLAID_CLIENT_ID,
+          secret: PLAID_SECRET_SANDBOX,
+          user: {
+            // This should correspond to a unique id for the current user.
+            client_user_id: 'uniqueId' + userId,
+          },
+          client_name: 'Pattern',
+          products,
+          country_codes: ['US'],
+          language: 'en',
+          webhook: httpTunnel.public_url + '/services/webhook',
+          transfer: {
+            intent_id: transferIntentId,
```

Comment:
Minor nit, but I feel like this would work better as an argument passed in to the function instead of relying on the  const in the larger scope